### PR TITLE
Snipets/compute_ip_address_get_vm_address

### PR DESCRIPTION
**Pull request by irudykss**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/client_library/ingredients/instances/ip_address/get_vm_address.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+# <INGREDIENT get_instance_ip_address>
+class IPType(Enum):
+    INTERNAL = "internal"
+    EXTERNAL = "external"
+
+
+def get_instance_ip_address(
+    instance: compute_v1.Instance, ip_type: IPType
```
Comment (msampathkumar) [inline compute/client_library/ingredients/instances/ip_address/get_vm_address.py:28]:
```suggestion
    instance: compute_v1.Instance, bool: internal_ip = False
```

This is a suggestion. If only available options are `internal` or `external`, then using bool datatype is simpler option.

Comment (msampathkumar) [reply compute/client_library/ingredients/instances/ip_address/get_vm_address.py:28]: Note: If you use bool input then `class IPType` is not required.

File: compute/client_library/snippets/instances/ip_address/get_vm_address.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+# directory and apply your changes there.
+
+
+# [START compute_ip_address_get_vm_address]
+from enum import Enum
+
+from google.cloud import compute_v1
+
+
+def get_instance(project_id: str, zone: str, instance_name: str) -> compute_v1.Instance:
```
Comment (msampathkumar) [inline compute/client_library/snippets/instances/ip_address/get_vm_address.py:28]: A question. Does this function work if VM has IPv6 address ?

File: compute/client_library/snippets/tests/test_ip_address.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        instance_name,
+    )
+    try:
+        internal_ip = get_instance_ip_address(instance, IPType.INTERNAL)
+        assert internal_ip == instance.network_interfaces[0].network_i_p
+    finally:
+        delete_instance(PROJECT, INSTANCE_ZONE, instance_name)
+
+
+def test_get_instance_external_ip_address():
```
Comment (msampathkumar) [inline compute/client_library/snippets/tests/test_ip_address.py:49]: Sugg: Creating instances & removing is time taking operations. This test function can also include test for both internal & external IP.

File: compute/client_library/snippets/instances/ip_address/get_vm_address.py (hunk 2 above)
Comment (irudykss) [reply compute/client_library/snippets/instances/ip_address/get_vm_address.py:28]: It is good point. Is doesn't, so I restored ip types to add v6.

File: compute/client_library/ingredients/instances/ip_address/get_vm_address.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    Args:
+        instance (compute_v1.Instance): instance to get
+        ip_type (IPType): The type of IP address to retrieve (ipv6, internal or external).
+
+    Returns:
+        str: The requested IP address of the instance.
+    """
+
+    if instance.network_interfaces:
```
Comment (m-strzelczyk) [inline compute/client_library/ingredients/instances/ip_address/get_vm_address.py:40]: Please avoid unnecessary indentation/nesting.

```py
if not instance.network_interfaces:
    return ""
# No indentation needed after this.
```

File: compute/client_library/ingredients/instances/ip_address/get_vm_address.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+
+# <INGREDIENT get_instance_ip_address>
+class IPType(Enum):
+    INTERNAL = "internal"
+    EXTERNAL = "external"
+    IP_V6 = "ipv6"
+
+
+def get_instance_ip_address(instance: compute_v1.Instance, ip_type: IPType) -> str:
```
Comment (m-strzelczyk) [inline compute/client_library/ingredients/instances/ip_address/get_vm_address.py:28]: What if the instance has multiple interfaces and addresses? We should return a list.

File: compute/client_library/snippets/tests/test_ip_address.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+from ..instances.delete import delete_instance
+from ..instances.ip_address.get_vm_address import get_instance_ip_address, IPType
+
+PROJECT = google.auth.default()[1]
+REGION = "us-central1"
+INSTANCE_ZONE = "us-central1-b"
+
+
+def test_get_instance_external_ip_address():
```
Comment (m-strzelczyk) [inline compute/client_library/snippets/tests/test_ip_address.py:33]: We should use fixtures for easier resource management.

- disk_fixture
- instance_fixture(disk_fixture)

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]

File: compute/client_library/ingredients/instances/ip_address/get_vm_address.py (hunk 5 above)
Comment (irudykss) [reply compute/client_library/ingredients/instances/ip_address/get_vm_address.py]: good point, fixed

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**